core/src/main/scala/caliban/GraphQLRequest.scala

```diff
@@ -26,3 +29,16 @@ private object GraphQLRequestCirce {
     } yield GraphQLRequest(query, operationName, variables)
 
 }
+
+private object GraphQLRequestPlayJson {
+  import play.api.libs.json._
+  import play.api.libs.json.Reads._
+  import play.api.libs.functional.syntax._
+
+  val graphQLRequestReads: Reads[GraphQLRequest] =
+    (
+      (JsPath \ "query").read[String] and
+        (JsPath \ "operationName").readNullable[String] and
+        (JsPath \ "variables").readNullable[Map[String, InputValue]]
+    )(GraphQLRequest.apply _)
```

Comment:
Why can't the macro be used here? It should be able to derive these already.

Reply:
fixed